web-services/client/src/main/java/datawave/webservice/query/QueryParametersImpl.java

```diff
@@ -89,11 +110,21 @@ public void validate(Map<String,List<String>> parameters) throws IllegalArgument
                 throw new IllegalArgumentException("Unknown condition.");
             }
         }
-        Preconditions.checkNotNull(this.query, "Query string cannot be null");
-        Preconditions.checkNotNull(this.queryName, "Query name cannot be null");
-        Preconditions.checkNotNull(this.persistenceMode, "Persistence mode cannot be null");
-        Preconditions.checkNotNull(this.auths, "Query auths cannot be null");
-        Preconditions.checkNotNull(this.expirationDate, "Expiration date cannot be null");
+        
+        try {
+            Preconditions.checkNotNull(this.query, "QueryParameter 'query' cannot be null");
+            Preconditions.checkNotNull(this.queryName, "QueryParameter 'queryName' cannot be null");
+            Preconditions.checkNotNull(this.persistenceMode, "QueryParameter 'persistence' mode cannot be null");
+            Preconditions.checkNotNull(this.auths, "QueryParameter 'auths' cannot be null");
+            Preconditions.checkNotNull(this.pagesize, "QueryParameter 'pagesize' cannot be null");
+            Preconditions.checkNotNull(this.pageTimeout, "QueryParameter 'pageTimeout' cannot be null");
+            Preconditions.checkNotNull(this.expirationDate, "QueryParameter 'expiration' date cannot be null");
+            Preconditions.checkNotNull(this.beginDate, "QueryParameter 'begin' cannot be null");
+            Preconditions.checkNotNull(this.endDate, "QueryParameter 'end' cannot be null");
+            Preconditions.checkNotNull(this.logicName, "QueryParameter 'logicName' cannot be null");
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e.getMessage());
+        }
```

Comment:
Based on the validate method here and the QueryExecutorBean it appears that all these parameters need to be present. I think it makes more sense to throw an IllegalArgumentException instead of letting the query process a little later and throwing a DatawaveWebApplicationException. If someone has another opinion please share here.

Reply:
Btw, I noticed that web service responses no longer seem to indicate the missing required parameter for the user the way that they used to, but I checked and that's also the case with the master branch behavior, so nothing that has to be addressed here. I'll create a separate ticket and see if I can figure out when/how that changed